spec/requests/api/v1/users_request_spec.rb

```diff
@@ -100,4 +100,46 @@
       expect { user.reload }.to change(user, :reset_password_token)
     end
   end
+
+  describe "DELETE /api/v1/users/destroy_self" do
+    context "when user unauthenticated" do
+      subject(:request_destroy_self) { delete "/api/v1/users/destroy_self" }
+
+      it "returns unauthorized status code" do
+        request_destroy_self
+        expect(response).to have_http_status(:unauthorized)
+      end
+
+      it "returns invalid_token message error" do
+        request_destroy_self
+        expect(response.parsed_body).to include({ error: "invalid_token" })
+      end
+    end
+
+    context "when user autheticated" do
```

Comment:
this context should have two contexts `when passowrd is valid` and `when password is invalid`